doc/source/whatsnew/v1.1.0.rst

```diff
@@ -224,6 +224,9 @@ Performance improvements
 - The internal index method :meth:`~Index._shallow_copy` now copies cached attributes over to the new index,
   avoiding creating these again on the new index. This can speed up many operations that depend on creating copies of
   existing indexes (:issue:`28584`, :issue:`32640`, :issue:`32669`)
+- Performance improvement when creating sparse :class:`DataFrame` from
+  ``scipy.sparse`` matrices using the :meth:`DataFrame.sparse.from_spmatrix`
+  constructor (:issue:`32196`).
```

Comment:
Added PRs by Joris, there have been 5 PRs on this in total.

Ideally another what's new entry should be added since PR's by @jorisvandenbossche  also made initialization of extension arrays faster under some conditions, as far as I understand. Though I would rather not add it here, nor am I competent on accurately formulating it.